backend/src/supabase/functions/room-jwtprovider/index.ts

```diff
@@ -1,8 +1,18 @@
 import { serve } from 'https://deno.land/std@0.131.0/http/server.ts'
+import { createClient, SupabaseClient } from 'https://esm.sh/@supabase/supabase-js@2';
 import { corsHeaders } from '../_shared/cors.ts';
 import Mux from "https://esm.sh/v106/@mux/mux-node@7.0.0/es2022/mux-node.js";
 
-async function generateToken(spaceId: string) {
+async function generateToken(supabaseClient: SupabaseClient, spaceId: string) {
+  const spacesId = await supabaseClient.from('rooms').select('providerId').eq('providerId', spaceId);
+
+  if(spacesId.data && spacesId.data.length === 0){
+    return new Response(JSON.stringify({ error: 'Wrong Space ID'}), {
+        headers: { ...corsHeaders, 'Content-Type': 'application/json' },
+        status: 400,
+      });
+  }
```

Comment:
I'd also wrap this in a `try..catch`, since we're calling a different service here.

```suggestion
try {
  const spacesId = await supabaseClient.from('rooms').select('providerId').eq('providerId', spaceId);

  if(spacesId.data && spacesId.data.length === 0){
    return new Response(JSON.stringify({ error: 'Wrong Space ID'}), {
      headers: { ...corsHeaders, 'Content-Type': 'application/json' },
      status: 400,
    });
  }
} catch (error) {
  return new Response(JSON.stringify({ error }), {
    headers: { ...corsHeaders, 'Content-Type': 'application/json' },
    status: 400,
  });
}
```